### PR TITLE
[ogre] Fix include path in usage

### LINKED ISSUE #15333: [ogre] Cannot open include file
****

**Is your feature request related to a problem? Please describe.**
when downloading ogre with vcpkg
`./vcpkg install ogre`
 all of the header files are not correctly included, they are all included with quotations "" rather than with brackets <> and the location of the header files are not accurate either. 

for example:
a header in Ogre.h will have this by default
`#include "OgreAnimation.h"`
when it should have
`#include <OGRE/OgreAnimation.h>`

**Proposed solution**
i can rewrite hundreds of header files manually to maybe get ogre to work but a update may break the changes...

**Describe alternatives you've considered**
i'm very new to c++ and i just wanted to dabble with it, sorry...
**Additional context**
i am running on windows 10

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #15333

   Replace #include "OgreRTShaderSystem.h" as #include "../RTShaderSystem/OgreRTShaderSystem.h" in ../Bites/OgreSGTechniqueResolverListener.h file.

Note: No need to test features.

### CODE REVIEW

File: ports/ogre/portfile.cmake [hunk 1]
```diff
@@ -74,57 +69,59 @@ vcpkg_configure_cmake(
...
     endif()
     if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "Debug")
-        file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link)
+        file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/debug/lib/manual-link")
         if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
-            file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/OgreMain_d.lib ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link/OgreMain_d.lib)
+            file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/OgreMain_d.lib" "${CURRENT_PACKAGES_DIR}/debug/lib/manual-link/OgreMain_d.lib")
         else()
-            file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/OgreMainStatic_d.lib ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link/OgreMainStatic_d.lib)
+            file("RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/OgreMainStatic_d.lib" "${CURRENT_PACKAGES_DIR}/debug/lib/manual-link/OgreMainStatic_d.lib")
```
Comment (NancyLi1013) [inline ports/ogre/portfile.cmake:110]:
```suggestion
            file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/OgreMainStatic_d.lib" "${CURRENT_PACKAGES_DIR}/debug/lib/manual-link/OgreMainStatic_d.lib")
```

File: versions/o-/ogre.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "19babe49543ff790585bbc026326fe05446f9ea7",
```
Comment (NancyLi1013) [inline versions/o-/ogre.json:4]:
```suggestion
      "git-tree": "9c35ef4f5b30d305e6a128f9531692a64eae1b4a",
```

**Review by JackBoosY — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Depends on #19633

File: ports/ogre/portfile.cmake [hunk 3]
```diff
@@ -65,7 +61,6 @@ vcpkg_configure_cmake(
         -DOGRE_BUILD_RENDERSYSTEM_D3D11=ON
         -DOGRE_BUILD_RENDERSYSTEM_GL=ON
         -DOGRE_BUILD_RENDERSYSTEM_GL3PLUS=ON
```
Comment (BillyONeal) [inline ports/ogre/portfile.cmake:63]: Why did you remove:

```
-DOGRE_BUILD_RENDERSYSTEM_GLES=OFF
```

?

File: ports/ogre/portfile.cmake [hunk 4]
```diff
@@ -74,57 +69,59 @@ vcpkg_configure_cmake(
...
-    file(GLOB SHARE_FILES ${CURRENT_PACKAGES_DIR}/share/ogre/*.cmake)
+    file(GLOB SHARE_FILES "${CURRENT_PACKAGES_DIR}/share/ogre/*.cmake")
     foreach(SHARE_FILE ${SHARE_FILES})
         file(READ "${SHARE_FILE}" _contents)
         string(REPLACE "lib/OgreMain" "lib/manual-link/OgreMain" _contents "${_contents}")
         file(WRITE "${SHARE_FILE}" "${_contents}")
     endforeach()
 endif()
 
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/OGRE/Bites/OgreSGTechniqueResolverListener.h" "OgreRTShaderSystem.h" "../RTShaderSystem/OgreRTShaderSystem.h")
```
Comment (BillyONeal) [inline ports/ogre/portfile.cmake:122]: This looks like a bug in upstream, has it been reported there?

Comment (BillyONeal) [reply ports/ogre/portfile.cmake:122]: Also it seems unlikely that this is the only header affected; in the issue report the user says that they would have to edit "hundreds of header files", not only this one. I'm not positive what the correct fix is but I don't think this is.

**Review by BillyONeal — CHANGES_REQUESTED** [review]

File: ports/ogre/portfile.cmake (hunk 3 above)
Comment (NancyLi1013) [reply ports/ogre/portfile.cmake:63]: Since this option doesn't exist on the current source codes.

File: ports/ogre/portfile.cmake (hunk 4 above)
Comment (NancyLi1013) [reply ports/ogre/portfile.cmake:122]: Thanks for pointing this out. Since this only affects the usage in the project. I'm not sure if this is a upstream bug. In user's project sample, there is only one place that needs to update. For other header files, it is due to the user didn't include the header path correctly in his project. 

The same kinds of problem may be also in other header files. I also have no better way to solve this. I'll report this problem to upstream.

Comment (NancyLi1013) [reply ports/ogre/portfile.cmake:122]: upstream issue https://github.com/OGRECave/ogre/issues/2172

**Comment by NancyLi1013** [comment]
Closing this PR since this is described as expected behavior by Ogre maintainers. It's unnecessary to handle this in vcpkg side now. Please see https://github.com/OGRECave/ogre/issues/2172#issuecomment-910096941 and https://github.com/OGRECave/ogre/issues/2172#issuecomment-910523066.
